fluent-plugin-enhance-k8s-metadata/lib/sumologic/kubernetes/cache_strategy.rb

```diff
@@ -0,0 +1,30 @@
+module SumoLogic
+  module Kubernetes
+    # module for caching strategy
+    module CacheStrategy
+      require 'lru_redux'
+      require_relative 'reader.rb'
+
+      CACHE_TYPE_POD_LABELS = 'pod_labels'.freeze
+      CACHE_TYPE_OWNER_REFS = 'owner_refs'.freeze
+
+      def init_cache
+        @all_caches = {
+          CACHE_TYPE_POD_LABELS => LruRedux::TTL::ThreadSafeCache.new(@cache_size, @cache_ttl),
+          CACHE_TYPE_OWNER_REFS => LruRedux::TTL::ThreadSafeCache.new(@cache_size, @cache_ttl)
+        }
+      end
+
+      def get_pod_labels(namespace_name, pod_name)
+        key = "#{namespace_name}::#{pod_name}"
+        cache = @all_caches[CACHE_TYPE_POD_LABELS]
+        labels = cache[key]
+        if labels.nil?
+          labels = fetch_pod_labels(namespace_name, pod_name)
```

Comment:
A general question: it's a bit hard to follow this kind of function call in another module. Is it by convention people ignore the class reference? Would be much easier to follow if we know where doesn't that function come from. Also what if required modules overload a method, which one would be used?

Reply:
You can consider the module is another approach to do something like writing util functions in `trait` in Scala but more flexible. (e.g. [duck typing](https://stackoverflow.com/a/4205163/5472728)) It's more natural way to reuse code in Ruby rather than using inheritance.

For connivence, if you install Ruby plugin in VS code (and some dependency gems following the instruction), it can support Cmd+Click to jump between functions from different modules

Reply:
Gtk. Thanks!